stub_uploader/metadata.py

```diff
@@ -174,6 +177,44 @@ def verify_typeshed_req(req: Requirement) -> None:
 }
 
 
+def validate_response(resp: requests.Response, req: Requirement) -> None:
+    if resp.status_code != 200:
+        raise InvalidRequires(
+            f"Expected dependency {req} to be accessible on PyPI, but got {resp.status_code}"
+        )
+
+
+def get_sdist_requires(
+    sdist_data: dict[str, str], req: Requirement
+) -> Generator[Requirement, None, None]:
+    filename = sdist_data["filename"]
+    resp = requests.get(sdist_data["url"], stream=True)
+    validate_response(resp, req)
+    with open(filename, "wb") as file:
+        file.write(resp.raw.read())
+
+    folder_name = ""
+    with tarfile.open(filename) as file_in:
+        file_in.extractall()
+        for info in file_in:
+            folder_name = info.name
+            # Assume a single folder
+            break
+
+    requires_filepath = Path(folder_name, "*.egg-info", "requires.txt")
+    matches = glob(str(requires_filepath))
+    for match in matches:
+        lines: list[str] = []
+        with open(match) as requires_file:
+            lines = requires_file.readlines()
+        for line in lines:
+            # Skip empty lines and extras
+            if line[0] not in {"\n", "["}:
+                yield Requirement(line)
+        # Assume a single *.egg-info folder
```

Comment:
Same with the `# Assume a single folder` comment above. These are assumptions from what I saw testing a few distributions.
I could also check the length, give an error if 2 or more (then we can investigate and correctly deal with those cases).
And stop early if 0. Not sure what's the best practice for returning an empty Generator. `return`, `return iter(())`, `yield from ()` ?